### PR TITLE
Smooth Audio Playback

### PR DESCRIPTION
**Pull request by kelvinkirima014**
This PR is a first attempt at fixing #3. It introduces smooth audio playback using a single channel, i.e mono audio. Future work should extend this functionality to support stereo audio.

### CODE REVIEW

File: lib/common/ring.ts [hunk 1]
```diff
@@ -62,8 +62,14 @@ export class Ring {
 		const readPos = Atomics.load(this.state, STATE.READ_POS)
 		const writePos = Atomics.load(this.state, STATE.WRITE_POS)
 
+		let samplesPerFrame = 1
+		samplesPerFrame = frame.numberOfChannels
+
+		 const numberOfSamples = frame.numberOfFrames * samplesPerFrame
```
Comment (kixelated) [inline lib/common/ring.ts:68]: Multiplying by 1 for mono audio if I'm not mistaken? Same as the previous code?

File: lib/common/ring.ts [hunk 2]
```diff
@@ -62,8 +62,14 @@ export class Ring {
 		const readPos = Atomics.load(this.state, STATE.READ_POS)
 		const writePos = Atomics.load(this.state, STATE.WRITE_POS)
 
+		let samplesPerFrame = 1
+		samplesPerFrame = frame.numberOfChannels
```
[the remark this answers is not in the document]
Comment (kixelated) [reply lib/common/ring.ts:66]: The poor naming is partly my fault. AudioData uses `frame == sample` while I use `frames == group of samples`.

**Review by kixelated — CHANGES_REQUESTED** [review]
I don't understand the changes to the ring buffer. Does mono audio work without these those changes?

File: lib/common/ring.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kelvinkirima014) [reply lib/common/ring.ts]: Reverted to using the previous code as it works with mono audio and is a bit more dynamic with the channel count.

**Pull request closed without merging**